process/economics/economicsData.go

```diff
@@ -709,7 +709,7 @@ func (ed *economicsData) setGasLimitConfig(currentEpoch uint32) {
 		"minGasLimit", ed.minGasLimit,
 	)
 
-	ed.statusHandler.SetUInt64Value(common.MetricMaxGasPerTransaction, core.MinUint64(ed.maxGasLimitPerMiniBlock, ed.maxGasLimitPerMetaMiniBlock))
+	ed.statusHandler.SetUInt64Value(common.MetricMaxGasPerTransaction, core.MaxUint64(process.MaxGasLimitPerTxForSafeCrossShard, core.MinUint64(ed.maxGasLimitPerMiniBlock, ed.maxGasLimitPerMetaMiniBlock)))
```

Comment:
this is a little hard to understand for someone not involved in the discussions.

Reply:
I will add a comment in all the places where this calculation is used